### PR TITLE
chore: add testingType, initialLaunch to update check

### PR DESCRIPTION
**Pull request by kuceb**
- Closes TR-740

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

File: packages/server/lib/updater.js [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
+const _getManifest = ({ id, initialLaunch, testingType }) => {
   const url = konfig('desktop_manifest_url')
 
   return rp.get({
     url,
     headers: {
       'x-cypress-version': pkg.version,
+      'x-os-name': os.platform(),
       'x-machine-id': id,
+      'x-initial-launch:': initialLaunch,
```
Comment (flotwig) [inline packages/server/lib/updater.js:18]: headers are strings, i'm sure this is done somewhere down the line, but just to make sure the download server gets what it expects (either "true" or not "true"):

```suggestion
      'x-initial-launch:': String(initialLaunch),
```

File: packages/server/test/unit/updater_spec.js [hunk 2]
```diff
@@ -10,12 +10,15 @@ describe('lib/updater', () => {
...
 
-      Updater._getManifest('machine-id')
+      Updater._getManifest({ testingType: 'type', initialLaunch: true, id: 'machine-id' })
 
       expect(rp.get).to.be.calledWithMatch({
         headers: {
           'x-cypress-version': pkg.version,
+          'x-os-name': 'linux',
           'x-machine-id': 'machine-id',
+          'x-initial-launch:': true,
```
Comment (flotwig) [inline packages/server/test/unit/updater_spec.js:20]:
```suggestion
          'x-initial-launch:': 'true',
```

**Review by flotwig — DISMISSED** [review]
looks good after the string thing is merged

File: packages/desktop-gui/src/update/updates.js [hunk 3]
```diff
@@ -5,16 +5,23 @@ import updateStore from '../update/update-store'
 import ipc from '../lib/ipc'
 import { useLifecycle } from '../lib/use-lifecycle'
 
+let initialLaunch = true
```
Comment (flotwig) [inline packages/desktop-gui/src/update/updates.js:8]: hm, i had assumed that `initialLaunch` would be like... "is this the first time Cypress has been opened on this machine ID?", but it appears to be measuring "is this the first update check of a new Cypress application session?". is my understanding correct?

Comment (Bkucera) [reply packages/desktop-gui/src/update/updates.js:8]: correct. application session is desired behavior

File: packages/server/lib/updater.js (hunk 1 above)
Comment (Bkucera) [reply packages/server/lib/updater.js]: wait until you see the request-attempt header PR...

**Review by flotwig — APPROVED** [review]

**Pull request merged**